### ISSUE TITLE
Error load metadata for docker.io/library/alpine:3.12 with v3.0.1

### ISSUE THREAD

**Issue by puneri78** [docker/buildx#476]
This issue is specific to docker buildkit using --output command with docker build.

Using DOCKER_BUILDKIT=1 
Same dockerfile works fine with  v2.5.0.1

  - [x] I have tried with the latest version of my channel (Stable or Edge)

### Expected behavior
No error

### Actual behavior

=> [internal] load build definition from Dockerfile                                                                                        0.0s
 => => transferring dockerfile: 3.30kB                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                           0.0s
 => => transferring context: 2B                                                                                                             0.0s
 => ERROR [internal] load metadata for docker.io/library/alpine:3.12                                                                        4.5s
 => [internal] load metadata for docker.io/library/golang:1.15-alpine3.12                                                                   0.0s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                               0.0s

------
 > [internal] load metadata for docker.io/library/alpine:3.12:
------

### Information
  - macOS Version: 10.15.6 (Catalina)
  - Is it reproducible? - Yes
  - Is the problem new? - Yes
  - Did the problem appear with an update? - Yes, with v3.0.1 

### Steps to reproduce the behavior

Dockerfile  (just example)

Command : docker build --target test --output type=local,dest=$(pwd)/out --platform local  .

> FROM --platform=${BUILDPLATFORM}  golang:1.15-alpine3.12 AS base
> 
> WORKDIR /src
> 
> RUN apk add --no-cache openssh-client git
> 
> FROM alpine:3.12 AS test
>
> WORKDIR /app
>
> COPY --from=base xxxxx /app/
>

**Comment by black-snow** [comment]
Maybe related:

```
[+] Building 1.9s (3/3) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                         0.0s
 => => transferring dockerfile: 38B                                                                                                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                            0.0s
 => => transferring context: 34B                                                                                                                                                                                                                                             0.0s
 => ERROR [internal] load metadata for docker.io/library/tomcat:8.5-jdk8-openjdk-slim                                                                                                                                                                                        1.8s
------
 > [internal] load metadata for docker.io/library/tomcat:8.5-jdk8-openjdk-slim:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exec: "docker-credential-desktop.exe": executable file not found in $PATH, out: ``
```

**Comment by gangware72** [comment]
Anything on this?

**Comment by Slyke** [comment]
Also getting this error. Freshly installed docker on WSL2 and this error started happening after a few hours:
```
$ docker build --no-cache -t $FULL_NAME -f api.Dockerfile .
[+] Building 0.5s (2/3)
[+] Building 0.5s (3/3) FINISHED
 => [internal] load build definition from api.Dockerfile                                                                                                                                                                                                                                                                                             0.0s
 => => transferring dockerfile: 227B                                                                                                                                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                    0.0s
 => => transferring context: 34B                                                                                                                                                                                                                                                                                                                     0.0s
 => ERROR [internal] load metadata for docker.io/library/node:14                                                                                                                                                                                                                                                                                     0.4s
------
 > [internal] load metadata for docker.io/library/node:14:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exit status 1, out: 
```

Docker Version:
```
$ docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.7
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:17:34 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8891c58
  Built:            Mon Dec 28 16:15:28 2020
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Docker info:
```
$ docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)
  scan: Docker Scan (Docker Inc., v0.5.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 20.10.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.128-microsoft-standard
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 24.99GiB
 Name: docker-desktop
 ID: Q567:BLLG:6TEQ:4ITC:M3XT:OTF5:JGBI:H6AO:CVYV:G5MP:XXSH:ISZE
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio weight support
WARNING: No blkio weight_device support
WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Comment by pichkasik** [comment]
Worked for me:
1. Create an account on 'https://hub.docker.com/'
2. Run in terminal: 'docker login'
3. Authorized in terminal
4. Try run docker build again

**Comment by gangware72** [comment]
Thank you let me post back on this later in the day.

**Comment by sachwinster** [comment]
> Worked for me:
> 
> 1. Create an account on 'https://hub.docker.com/'
> 2. Run in terminal: 'docker login'
> 3. Authorized in terminal
> 4. Try run docker build again

Worked for me right away after this.

**Comment by evereq** [comment]
Happens to me today.
I was already logged in and so run `docker logout` first and next `docket login`. 
Thank you @pichkasik for a hint!

**Comment by greenbuks** [comment]
Worked for me:
1. Add Docker Desktop.exe to $PATH
2. Profit!

**Comment by ajithcravi** [comment]
Try 'sudo' if you are using linux

**Comment by renanleme** [comment]
> Try 'sudo' if you are using linux

This worked for me, thanks!
Using WSL2 on Windows.

**Comment by danielefranceschi** [comment]
Docker desktop 20.10.5 on WSL2, removing `"credsStore": "desktop.exe",` from `~/.docker/config.json` does the job (as usual).

**Comment by alfiedouglas-tewke** [comment]
`docker logout` & `docker login` did not work for me but after `sudo service docker restart` it worked.

**Comment by JunHCha** [comment]
> Docker desktop 20.10.5 on WSL2, removing `"credsStore": "desktop.exe",` from `~/.docker/config.json` does the job (as usual).

This works on me too. Thanks! :)

**Comment by mihail-salari** [comment]
For the ones for who none of the solutions from the above work, you could try this:
1. Run `docker pull ${dockerPkgLink}` for pulling the package which docker failed to load metadata for
2. Run docker build again

This worked for me

**Comment by route33** [comment]
For me, it worked only after I changed:
`FROM envoyproxy/envoy:v1.17.1`
to
`FROM docker.io/envoyproxy/envoy:v1.17.1`

**Comment by jackno** [comment]
Or even better, `rm /Users/<username>/.docker/.token_seed`

**Comment by gt-novelt** [comment]
Using WSL2, restarting docker4windows, restarting the wsl instances would not work.
Windows reboot fixed it for me 🤷

**Comment by Overfacto** [comment]
Docker Desktop update worked for me. 

It maybe related about Docker desktop version - docker hub communication trouble

**Comment by kotAPI** [comment]
try sudo on macOS as well

**Comment by pedrovsiqueira** [comment]
> For the ones for who none of the solutions from the above work, you could try this:
> 
> 1. Run `docker pull ${dockerPkgLink}` for pulling the package which docker failed to load metadata for
> 2. Run docker build again
> 
> This worked for me

This worked for me. Thanks.

**Comment by sheldonldev** [comment]
If nothing above help, you can check `Dockerfile` if there's something typo, such as a wrong version etc.

**Comment by chavangorakh1999** [comment]
> Worked for me:
> 
> 1. Create an account on 'https://hub.docker.com/'
> 2. Run in terminal: 'docker login'
> 3. Authorized in terminal
> 4. Try run docker build again

Worked for me too 🥰🥰

**Comment by sauravpradhan** [comment]
I am on latest docker for windows.
![image](https://user-images.githubusercontent.com/1622949/121516575-fe9bb380-ca0b-11eb-9152-7978585be668.png)

Docker build wont work for any dockerfile. This issue still persists. I have tried most of the solution from comments but of no help. Any suggestions.

**Comment by sauravpradhan** [comment]
Update: https://stackoverflow.com/questions/64382812/docker-has-the-same-error-regardless-of-what-i-try-to-build-windows-10 helped me get rid of the issue.

**Comment by michaelpierrelee** [comment]
> Update: https://stackoverflow.com/questions/64382812/docker-has-the-same-error-regardless-of-what-i-try-to-build-windows-10 helped me get rid of the issue.

This works for me,  I deactived buildkit following https://stackoverflow.com/a/64466563

I installed Docker desktop on Windows 10 Edu with Debian for WSL2. I wanted to build the app of the tutorial https://docs.docker.com/get-started/02_our_app/ with a terminal connected to my Debian. I got the error of this issue when I was on Debian, but not on Windows with the powershell.

I still don't know why it has worked.

**Comment by guoxiangke** [comment]
remove un-reachable registry-mirrors works for me.

**Comment by moracabanas** [comment]
> remove un-reachable registry-mirrors works for me.

Can you elaborate on where can you configure registry mirrors a little bit? 
Is this setting on /etc/docker/daemon.json? 
I am confused by registry-mirrors Docker documentation and I have makefiles from old DevOps team I can't reach anymore.
So a few of lines from you could save my day! 😅

**Comment by ahmedyarub** [comment]
In my case it was the proxy. I just disabled it in Docker Desktop and everything worked again.

**Comment by tadyen** [comment]
> 
> 
> > Update: https://stackoverflow.com/questions/64382812/docker-has-the-same-error-regardless-of-what-i-try-to-build-windows-10 helped me get rid of the issue.
> 
> This works for me, I deactived buildkit following https://stackoverflow.com/a/64466563
> 
> I installed Docker desktop on Windows 10 Edu with Debian for WSL2. I wanted to build the app of the tutorial https://docs.docker.com/get-started/02_our_app/ with a terminal connected to my Debian. I got the error of this issue when I was on Debian, but not on Windows with the powershell.
> 
> I still don't know why it has worked.

I ran into the issue with Debian for WSL2. I switched to powershell /terminal and it works. No clue either but your comment saved me!

**Comment by lonce** [comment]
How can they expect people to use docker if they make it so damn difficult just to pull an f*in image from the site. I had to identify busses, crosswalks, and bikes - then creat some creditial, then enable 2 factor identification, and I sill can't build a damn image on my machine. Sheesh.

**Comment by bbennya** [comment]
> This issue is specific to docker buildkit using --output command with docker build.
> 
> Using DOCKER_BUILDKIT=1
> Same dockerfile works fine with v2.5.0.1
> 
> * [x]  I have tried with the latest version of my channel (Stable or Edge)
> 
> ### Expected behavior
> No error
> 
> ### Actual behavior
> => [internal] load build definition from Dockerfile 0.0s
> => => transferring dockerfile: 3.30kB 0.0s
> => [internal] load .dockerignore 0.0s
> => => transferring context: 2B 0.0s
> => ERROR [internal] load metadata for docker.io/library/alpine:3.12 4.5s
> => [internal] load metadata for docker.io/library/golang:1.15-alpine3.12 0.0s
> => [auth] library/alpine:pull token for registry-1.docker.io 0.0s



for me it was a DNS issue.
I had the following error:

$ docker build .
[+] Building 11.3s (4/4) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                        0.0s
 => => transferring dockerfile: 38B                                                                                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                             0.0s
 => ERROR [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                     11.3s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                                               0.0s
------
 > [internal] load metadata for docker.io/library/alpine:latest:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: failed to authorize: rpc error: code = Unknown desc = failed to fetch oauth token: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Falpine%3Apull&service=registry.docker.io: dial tcp: lookup auth.docker.io on 172.30.208.1:53: read udp 172.30.217.138:43054->172.30.208.1:53: i/o timeout

solved by  adding a known DNS server:

 echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf > /dev/null

**Comment by udibagas** [comment]
same issue. thx

**Comment by Davekibh** [comment]
Guys I am having he following error please help me:

[+] Building 2.9s (3/3) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                            0.6s
 => => transferring dockerfile: 32B                                                                                                                                                             0.0s
 => [internal] load .dockerignore                                                                                                                                                               0.8s
 => => transferring context: 2B                                                                                                                                                                 0.0s
 => ERROR [internal] load metadata for davekibh.jfrog.io/docker/alpine:3.11.5                                                                                                                   1.7s
------
 > [internal] load metadata for davekibh.jfrog.io/docker/alpine:3.11.5:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: davekibh.jfrog.io/docker/alpine:3.11.5: not found

**Comment by crazy-max** [comment]
Closing this ticket because it's collecting many comments that are unrelated to the original issue but here is the state for some of them:

### `error getting credentials - err: exit status 1`

Should have been fixed since Docker Desktop 3.4.0:

* https://github.com/docker/for-win/issues/11261

### `err: exec: "docker-credential-desktop.exe": executable file not found in $PATH, out:`

Upstream issue, see:

* https://github.com/docker/for-win/issues/7957
* https://github.com/docker/for-win/issues/6652

If you still encounter this, please open an issue on https://github.com/docker/for-win

### `.token_seed: unexpected end of JSON input`

Fixed by moby/buildkit#2050. I will open a PR on [docker/cli](https://github.com/docker/cli/) 20.10 branch to update vendors (cc @thaJeztah).

**Comment by robertoschuh** [comment]
> 1. Run `docker pull ${dockerPkgLink}` for pulling the package which docker failed to load metadata for
> 2. Run docker build again

Yeah, it works for me! Thanks mate. Anyway I still trying to understand why this issue came suddenly. I am working on Windows 10 and WSL2 (Ubuntu 20.04).

**Comment by yotcap** [comment]
> Worked for me:
> 
> 1. Create an account on 'https://hub.docker.com/'
> 2. Run in terminal: 'docker login'
> 3. Authorized in terminal
> 4. Try run docker build again

veeeeeery thx :)

**Comment by LoicEsk** [comment]
Hello,

I had similar issue. For me, I found the fix here : https://stackoverflow.com/questions/65361083/docker-build-failed-to-fetch-oauth-token-for-openjdk

It works well by setting the buildkit to false in Docker settup.

**Comment by joshspicer** [comment]
+1, I was unblocked by setting `buildKit` to false

![image](https://user-images.githubusercontent.com/23246594/133831038-fb76bfe4-e431-422b-8381-4a96077559b5.png)

**Comment by joshspicer** [comment]
>  I think the failure may be expected, but the error message definitely could be improved. Could you open a new ticket instead? (of possible with a minimum reproducer)

Does this still seem like something expected? @thaJeztah

**Comment by josephlim75** [comment]
Same here, i have issue with buildkit.  Once I set `DOCKER_BUILDKIT=0` or change Docker Desktop preference to `buildkit: false`, everything work just fine.  If is `Dockerfile` issue, both settings should fail, but the problem seems to be more on buildkit

**Comment by DylanMashini** [comment]
> Or even better, `rm /Users/<username>/.docker/.token_seed`

this is the only thing that worked for me

**Comment by okechukwu-opara** [comment]
> docket login



> > Or even better, `rm /Users/<username>/.docker/.token_seed`
> 
> this is the only thing that worked for me

this worked for me. just on time

**Comment by kmturley** [comment]
In my case I was loading a Dockerfile:
```
FROM riscv64/debian:latest

WORKDIR /app
COPY . /app

RUN apt-get update && apt-get install build-essential cmake -y

CMD ["cmake --version"]
```

The error returned was:
```
ERROR [internal] load metadata for docker.io/riscv64/debian:latest
```

When visiting Docker hub, I noticed there was no tag called `latest`
https://hub.docker.com/r/riscv64/debian/tags

Changing to a valid tag fixed the issue:
```
FROM riscv64/debian:sid-slim

WORKDIR /app
COPY . /app

RUN apt-get update && apt-get install build-essential cmake -y

CMD ["cmake --version"]
```

Make sure the Docker image AND tag exist, otherwise you will recieve an error.

**Comment by moamdx** [comment]
> Worked for me:
> 
>     1. Create an account on 'https://hub.docker.com/'
> 
>     2. Run in terminal: 'docker login'
> 
>     3. Authorized in terminal
> 
>     4. Try run docker build again

that is worked for me
THX

**Comment by gustavorizzon** [comment]
> Worked for me:
> 
> 1. Create an account on 'https://hub.docker.com/'
> 2. Run in terminal: 'docker login'
> 3. Authorized in terminal
> 4. Try run docker build again

In my case, I was using Docker with WSL2 on Windows. I just tried to run `docker-compose up --build` with `sudo` and it worked without creating any credentials.

I hope this solution helps anyone else too. :)

**Comment by AllanCapistrano** [comment]
I just opened a new terminal window and it works for me

**Comment by henryejemuta** [comment]
> 2\. docker login


I already have a docker hub account but get the below error when trying to login from terminal
`Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: TLS handshake timeout`
But same details work well on the browser

**Comment by pierrebelin** [comment]
> 1. docker pull

It did the trick for me !
If you want to verify that the update worked, try to pull any image. It will display an error message if the config.json is incorrect

**Comment by luillyfe** [comment]
Maybe is getting the wrong platform to build. Specify it in the command: `docker build -t tag:1.0.0 --platform=linux/arm64 .`

**Comment by BerramouMohamed** [comment]
Hello, the commands below worked for me.

export DOCKER_BUILDKIT=0
export COMPOSE_DOCKER_CLI_BUILD=0

**Comment by AndyWronski** [comment]
> In my case it was the proxy. I just disabled it in Docker Desktop and everything worked again.

this workd for me .  M1 Macbook

**Comment by ikromjonxusanov** [comment]
In this case you did not download alpine:3.12 image, For working your code you should download alpine:3.12 image from docker hub,
docker pull alpine:3.12

**Comment by ghorbani-mohammad** [comment]
> For the ones for who none of the solutions from the above work, you could try this:
> 
> 1. Run `docker pull ${dockerPkgLink}` for pulling the package which docker failed to load metadata for
> 2. Run docker build again
> 
> This worked for me

I had this problem today and this solution worked for me.
Windows 10
WSL2

**Comment by anurag-sachan** [comment]
_-> Tried removing docker config file.
-> Tried pulling & using different docker images & versions respectively.
It didn't resolved the issue._

[+] Building 3.7s (4/4) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 36B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => ERROR [internal] load metadata for docker.io/library/java:8            3.6s
 => [auth] library/java:pull token for registry-1.docker.io                0.0s
------
 > [internal] load metadata for docker.io/library/java:8:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/library/java:8: not found.

**Comment by thaJeztah** [comment]
@anurag-sachan I think the `java` repository (and the `openjdk` image that it redirects to) have been deprecated, so that image does not (no longer) exists; https://github.com/docker-library/openjdk/issues/505

```bash
 docker manifest inspect docker.io/library/java:8
no such manifest: docker.io/library/java:8
```

**Comment by anurag-sachan** [comment]
> @anurag-sachan I think the `java` repository (and the `openjdk` image that it redirects to) have been deprecated, so that image does not (no longer) exists; [docker-library/openjdk#505](https://github.com/docker-library/openjdk/issues/505)
> 
> ```shell
>  docker manifest inspect docker.io/library/java:8
> no such manifest: docker.io/library/java:8
> ```

Yeah i tried different ways, so it works with _docker pull openjdk_ now.
Thank You!

**Comment by thaJeztah** [comment]
Be aware that the `openjdk` repository is also deprecated (the page on Docker Hub shows alternatives; https://hub.docker.com/_/openjdk)

**Comment by anurag-sachan** [comment]
> Be aware that the `openjdk` repository is also deprecated (the page on Docker Hub shows alternatives; https://hub.docker.com/_/openjdk)

Yup, I am using the same. Thanks for the help!

**Comment by daneshwari-858** [comment]
![Screenshot (9)](https://user-images.githubusercontent.com/102403971/201902866-ab7239a5-41c0-4f8a-880a-4870cc99f44b.png)





i am getting this what how can i resolve this

**Comment by thaJeztah** [comment]
@daneshwari-858 It looks like you used `COPY`, which means you're trying to copy a file named `apk` into the image; 

```dockerfile
COPY apk add --update reactjs
```

You probably meant to use `RUN`;

```dockerfile
RUN apk add --update reactjs
```

When using `apk` to install packages, I would also recommend using `--no-cache` instead of `--update`; the `--no-cache` option also makes sure that `apk`'s package index is up-to-date, but (unlike `--update`) doesn't store the updated package index files, so prevents making your image larger by keeping those around;

```dockerfile
RUN apk add --no-index reactjs
```

**Comment by daneshwari-858** [comment]
![Screenshot_2022-11-15-18-01-14-49_c37d74246d9c81aa0bb824b57eaf7062](https://user-images.githubusercontent.com/102403971/201920335-0f3cf636-2b23-4444-aa1a-
45a488c769da.jpg)



How can I do this bit confusing. Donno coding please help me

**Comment by thaJeztah** [comment]
@daneshwari-858 you need to edit your Dockerfile (usually the file is named `Dockerfile` as well)

**Comment by daneshwari-858** [comment]
![Screenshot (4)](https://user-images.githubusercontent.com/102403971/201925292-edd5578e-bef6-4436-a036-b60224bc7708.png)



how can i do this please me with a code.
i am very gland that you are replaying me.
thanks

**Comment by thaJeztah** [comment]
@daneshwari-858 sorry, but that's really out of scope for this ticket; the "build" section in the docker documentation would be a starting point to understand how "docker build" works; https://docs.docker.com/build/, but otherwise, I see from your earlier screenshot that you were following a training from, and it's better to contact them to improve the training material if it's not clear.

**Comment by daneshwari-858** [comment]
yes sir, i am  not getting enough help from that.but thats ok you are made an idea in my mind tq for that.
please can you say y this error is coming while installing jenkins.
tell me please.
![Screenshot (7)](https://user-images.githubusercontent.com/102403971/201928887-0847a82e-7d6b-4c2f-8aff-70ff094b3ce6.png)

**Comment by VukasinVujicVegaIT** [comment]
> Worked for me:
> 
> 1. Create an account on 'https://hub.docker.com/'
> 2. Run in terminal: 'docker login'
> 3. Authorized in terminal
> 4. Try run docker build again

this helped me how ever I could log in right away so I google again some guy said to delete "credsStore" : "desktop", inside "~/.docker/config.json"

**Comment by AndresCarrasquilla96** [comment]
> For the ones for who none of the solutions from the above work, you could try this:
> 
>     1. Run `docker pull ${dockerPkgLink}` for pulling the package which docker failed to load metadata for
> 
>     2. Run docker build again
> 
> 
> This worked for me

My error was:
```
 => ERROR [... internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                               
------
 > [... internal] load metadata for docker.io/library/debian:buster-slim:
------
failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exit status 1, out: ``
```
And thats what fixed it for me.

**Comment by arnold-plxr** [comment]
> In my case I was loading a Dockerfile:
> 
> ```
> FROM riscv64/debian:latest
> 
> WORKDIR /app
> COPY . /app
> 
> RUN apt-get update && apt-get install build-essential cmake -y
> 
> CMD ["cmake --version"]
> ```
> 
> The error returned was:
> 
> ```
> ERROR [internal] load metadata for docker.io/riscv64/debian:latest
> ```
> 
> When visiting Docker hub, I noticed there was no tag called `latest` https://hub.docker.com/r/riscv64/debian/tags
> 
> Changing to a valid tag fixed the issue:
> 
> ```
> FROM riscv64/debian:sid-slim
> 
> WORKDIR /app
> COPY . /app
> 
> RUN apt-get update && apt-get install build-essential cmake -y
> 
> CMD ["cmake --version"]
> ```
> 
> Make sure the Docker image AND tag exist, otherwise you will recieve an error.


This method worked for me!

**Comment by Lucasalmeida-ux** [comment]
work for me 
`docker pull node:14-alpine` before

**Comment by tvcapital** [comment]
> work for me `docker pull node:14-alpine` before

thanks, worked for me

**Comment by mehul1011** [comment]
> Docker desktop 20.10.5 on WSL2, removing `"credsStore": "desktop.exe",` from `~/.docker/config.json` does the job (as usual).

This would probably work for 90% of people.

**Comment by alexravenna** [comment]
> Check your build stages.
> 
> I inserted and then removed a stage, but forgot to update the "FROM" tag.
> 
> So for me, this was the problem:
> 
> ```
> FROM golang:1.17 as base
> FROM build as dev
> ```
> 
> This should be:
> 
> ```
> FROM golang:1.17 as base
> FROM base as dev
> ```

I received this error after having the same exact issue as you...it took me forever to find the culprit until I opened the beta "Builds" view in Docker Desktop and finally saw the error highlighted in the Dockerfile:

![image](https://github.com/docker/buildx/assets/29532881/bc2a64c4-1d7d-46aa-92d7-b17fa49db4c3)
`base` doesn't exist in the Dockerfile; `build` is what I needed😅 

It would be nice if the error tracing on the CLI could be a bit more specific in cases like this so that we wouldn't have to go digging through GitHub, Stack Overflow and then _happen_ to activate a beta feature in order to see what`s _truly_ going on.

**Comment by crazy-max** [comment]
@alexravenna 

That does not look related to this thread and I have the same behavior in cli and new build UI:

```dockerfile
FROM golang:1.17 as base
FROM build as dev
```

```
$ docker buildx build .
...
#3 [internal] load metadata for docker.io/library/build:latest
#3 ERROR: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
------
 > [internal] load metadata for docker.io/library/build:latest:
------
Dockerfile:2
--------------------
   1 |     FROM golang:1.17 as base
   2 | >>> FROM build as dev
   3 |
--------------------
ERROR: failed to solve: build: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
```

![image](https://github.com/docker/buildx/assets/1951866/15d3e2df-1f5d-444c-927b-16e382a1a549)

Feel free to open a new issue if that's not the case for you.

I'm locking this issue since this has been solved in Docker Desktop.
